plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQuerySplitManager.java

```diff
@@ -161,10 +163,12 @@ private List<BigQuerySplit> readFromBigQuery(ConnectorSession session, TableDefi
     }
 
     @VisibleForTesting
-    ReadSession createReadSession(ConnectorSession session, TableId remoteTableId, List<String> projectedColumnsNames, Optional<String> filter, int actualParallelism)
+    ReadSession createReadSession(ConnectorSession session, TableId remoteTableId, List<String> projectedColumnsNames, Optional<String> filter)
     {
         ReadSessionCreator readSessionCreator = new ReadSessionCreator(bigQueryClientFactory, bigQueryReadClientFactory, viewEnabled, arrowSerializationEnabled, viewExpiration, maxReadRowsRetries);
-        return readSessionCreator.create(session, remoteTableId, projectedColumnsNames, filter, actualParallelism);
+        // At least 100 to cater for cluster scale up
+        int desiredParallelism = Math.min(nodeManager.getRequiredWorkerNodes().size() * 3, 100);
```

Comment:
@findepi Did you mean to clamp this to 100 at-least? Current code is `min(worker * 3, 100)` which can be less than 100.
Did you mean `max(100, min(worker * 3, 100))`?

Reply:
Notice that the ReadSession creation takes two parameters - a maximal number of streams, and a preferred minimal number of streams, which the API tries to accommodate but treats it as a recommendation only, not binding limit. This logic is similar to what the Spark BigQuery connector is doing. 

By the way, why not use the bigquery-connector-common library used by both Spark and Hive connectors?

Reply:
@davidrabinowitz  that you for your input!

> By the way, why not use the bigquery-connector-common library used by both Spark and Hive connectors?

I think this is a very important question, but I definitely am not in a position to answer it. Please allow me to gloss over it.

> Notice that the ReadSession creation takes two parameters - a maximal number of streams, and a preferred minimal number of streams, which the API tries to accommodate but treats it as a recommendation only, not binding limit. 

Indeed. And this PR switches us from setting the max, to setting the min.
The idea is that we actually don't want to have a static max. For example, if the data set is very large, we want to have a very large number of "reasonably sized" splits. 
However, I don't know whether this is how it works.

@davidrabinowitz do we need to set both 'recommended min' and 'strict max' parameters? or can we just set the 'recommended min' ?

Reply:
Spark BigQuery connector seems to set both (https://github.com/search?q=repo%3AGoogleCloudDataproc%2Fspark-bigquery-connector+setPreferredMinStreamCount&type=code).

Setting max might be useful for example if we want to avoid having too many splits.

Reply:
Sure, but what should be the max value? 10k?
if the actual limit is 1k, we're fine.

Reply:
> By the way, why not use the bigquery-connector-common library used by both Spark and Hive connectors?

@davidrabinowitz @anoopj I sent DM in Trino community Slack. Let's continue the discussion there.